rasa/nlu/selectors/diet_selector.py

```diff
@@ -0,0 +1,427 @@
+        # number of epochs
+        EPOCHS: 300,
+        # set random seed to any int to get reproducible results
+        RANDOM_SEED: None,
+        # optimizer
+        LEARNING_RATE: 0.001,
+        # embedding parameters
+        # default dense dimension used if no dense features are present
+        DENSE_DIM: {TEXT: 512, LABEL: 512},
+        # dimension size of embedding vectors
+        EMBED_DIM: 20,
+        # the type of the similarity
+        NUM_NEG: 20,
+        # flag if minimize only maximum similarity over incorrect actions
+        SIMILARITY_TYPE: "auto",  # string 'auto' or 'cosine' or 'inner'
+        # the type of the loss function
+        LOSS_TYPE: "softmax",  # string 'softmax' or 'margin'
+        # number of top responses to normalize scores for softmax loss_type
+        # set to 0 to turn off normalization
+        RANKING_LENGTH: 10,
+        # how similar the algorithm should try
+        # to make embedding vectors for correct intent labels
+        MAX_POS_SIM: 0.8,  # should be 0.0 < ... < 1.0 for 'cosine'
+        # maximum negative similarity for incorrect intent labels
+        MAX_NEG_SIM: -0.4,  # should be -1.0 < ... < 1.0 for 'cosine'
+        # flag: if true, only minimize the maximum similarity for
+        # incorrect intent labels
+        USE_MAX_NEG_SIM: True,
+        # scale loss inverse proportionally to confidence of correct prediction
+        SCALE_LOSS: True,
+        # regularization parameters
+        # the scale of L2 regularization
+        REGULARIZATION_CONSTANT: 0.002,
+        # the scale of how critical the algorithm should be of minimizing the
+        # maximum similarity between embeddings of different intent labels
+        NEG_MARGIN_SCALE: 0.8,
+        # dropout rate for rnn
+        DROPRATE: 0.2,
+        # dropout rate for attention
+        DROPRATE_ATTENTION: 0,
+        # use a unidirectional or bidirectional encoder
+        UNIDIRECTIONAL_ENCODER: False,
+        # if true apply dropout to sparse tensors
+        SPARSE_INPUT_DROPOUT: False,
+        # visualization of accuracy
+        # how often to calculate training accuracy
+        EVAL_NUM_EPOCHS: 20,  # small values may hurt performance
+        # how many examples to use for calculation of training accuracy
+        EVAL_NUM_EXAMPLES: 0,  # large values may hurt performance,
+        # if true random tokens of the input message will be masked and the model
+        # should predict those tokens
+        MASKED_LM: False,
+        # if true use key relative embeddings in attention
+        KEY_RELATIVE_ATTENTION: False,
+        # if true use key relative embeddings in attention
+        VALUE_RELATIVE_ATTENTION: False,
+        # max position for relative embeddings
+        MAX_RELATIVE_POSITION: None,
+        # selector config
+        # name of the intent for which this response selector is to be trained
+        "retrieval_intent": None,
```

Comment:
I don't think constants are a great idea here in general, but not having this uniform is worse :D

Reply:
We introduced the constant to make it easier in the code itself to refer to the configuration as we use them quite often. We add a constant for the `retrieval_intent`.

Reply:
I agree constants are questionable, but we reuse these string keys in many places throughout the code. alternative would be to create HParams class, and populate it with config values. Then we'd use these strings only ones

Reply:
as long as it is consistent and all of them are constant I am good :+1: